packages/universe-components/src/profile/ProfileForm.tsx

```diff
@@ -94,21 +59,68 @@ export default async function Profile({
   const stories = userData.stories;
   const nopic = '/media/nopic.png';
 
+  async function saveImage(file: File) {
+    console.log('IMAGE', file)
+    //if(file){ return {error:'no image provided'} }
+    const name = randomString(10)
+    const body = new FormData()
+    body.append('name', name)
+    body.append('folder', 'avatars')
+    body.append('file', file)
+    const resp = await fetch('/api/upload', { method: 'POST', body })
+    const result = await resp.json()
+    return result
+  }
+
+  // Action to handle form submission
+  async function handleSaveProfile(formData: FormData, userId: string) {
+    const file = formData.get('file') as File | null;
+    const name = formData.get('name') as string;
+    const email = formData.get('email') as string;
+
+    console.log(name, email);
+    console.log({ file });
+    let image = (formData.get('currentImage') as string) ?? '';
+    if (file) {
+      //const fileUploadResponse = await uploadFile({ file, name, folder: 'avatars' });
+      const fileUploadResponse = await saveImage(file);
+      if (fileUploadResponse.success) {
+        image = fileUploadResponse.url ?? '';
+      }
+    }
+    //const updatePayload = { name, email, image };
+    //const updateData = await setUser(userId, updatePayload);
+
+    //if (!updateData) {
+    //  throw new Error('Error updating user data');
+    //}
+
+    const data = { name, email, image }
+    console.log('USER', data)
+    const res = await fetch(`/api/profile/${userId}`,{method:'post', body:JSON.stringify(data)})
```

Comment:
server action wasn't working?

Reply:
I don't remember now but for sure it was giving me headaches and had to go the direct route. We may try again when adding impact to profile. TODO.